### PR TITLE
Correct a variety of clippy lints

### PR DESCRIPTION
**Pull request by kinnison**
Deal with a number of clippy lints

### CODE REVIEW

File: tests/mock/dist.rs [hunk 1]
```diff
@@ -414,7 +414,7 @@ impl MockDistServer {
                 .iter()
                 .map(|v| toml::Value::String((**v).to_owned()))
                 .collect();
-            toml_profiles.insert(profile.to_string(), toml::Value::Array(array));
+            toml_profiles.insert((*profile).to_string(), toml::Value::Array(array));
```
Comment (tesuji) [inline tests/mock/dist.rs:417]: What about using String::from?

Comment (kinnison) [reply tests/mock/dist.rs:417]: Probably nothing, I'll admit I went with the recommendation from the compiler :D

Comment (tesuji) [reply tests/mock/dist.rs:417]: So str::to_string() use String::from under the hood: https://doc.rust-lang.org/nightly/src/alloc/string.rs.html#2179-2184.
I would file an issue in clippy in this case.

Comment (kinnison) [reply tests/mock/dist.rs:417]: I'll leave myself a TODO unless you want to do it.

Comment (tesuji) [reply tests/mock/dist.rs:417]: I don't bother me that much. Just a style.

Comment (kinnison) [reply tests/mock/dist.rs:417]: Fair enough, I'll bear it in mind next time clippy talks to me in that way :D

**Pull request merged**